app/controllers/concerns/effective_user.rb

```diff
@@ -1,7 +1,15 @@
 module EffectiveUser
   def effective_user
     return current_user if effective_user_id == current_user&.id
-    return User.find_by(id: effective_user_id) if effective_user_id
+
+    user = User.find_by(id: effective_user_id) if effective_user_id
+
+    if user.nil?
+      session.delete(:doc_capture_user_id)
```

Comment:
I find it a little strange we delete this after-the-fact, and as a side-effect of a getter method.

Should we delete this at the time that it becomes invalid? As in your pull request comment: "If the user deletes their account".

Reply:
Yeah, we considered that at first. But I think keeping it here means that things are slightly less "spread out" and easier to keep track of. Hopefully it will be academic because I am trying to get a ticket to remove `effective_user` refined and on Team Ada's plate.